sklearn/model_selection/_search.py

```diff
@@ -968,7 +968,10 @@ def _store(key_name, array, weights=None, splits=False, rank=False):
                 # when input is nan, scipy >= 1.10 rankdata returns nan. To
                 # keep previous behaviour nans are set to be smaller than the
                 # minimum value in the array before ranking
-                min_array_means = min(array_means) - 1
+                min_array_means = np.nanmin(array_means) - 1
```

Comment:
When `array_means` is all nan, there is a RuntimeWarning:

```
RuntimeWarning: All-NaN slice encountered
```

Reply:
We can either catch it or check for all nans first before calling `nanmin`.

Reply:
I catch this case early.